File: src/chapter1/GTCount.java
package chapter1;

/**
 * Created on 16/5/8.
 * author: robinjia
 * email: dengshilong1988@gmail.com
 */
public class GTCount {
    public static int gtCount(int[] f, int[] g) {
        int i = 0;
        int j = 0;
        int result = 0;
        while (i < f.length && j < g.length) {
            if (f[i] > g[j]) {
                result += f.length - i;
                j++;
            } else {
                i++;
            }
        }
        return result;
    }
    public static void main(String[] args) {
        int[] f = {1, 3, 5, 7, 9};
        int[] g = {2, 3, 4, 7, 8};
        System.out.println(gtCount(f, g));
    }
}

File: chapter1/gt_count.py
# coding: utf-8


def gt_count(f, g):
    i = 0
    j = 0
    result = 0
    while i < len(f) and j < len(g):
        if f[i] > g[j]:
            result += len(f) - i
            j += 1
        else:
            i += 1
    return result


if __name__ == "__main__":
    f = [1, 3, 5, 7, 9]
    g = [2, 3, 4, 7, 8]
    print(gt_count(f, g))

